src/components/asynccontent/AsyncContent.js

```diff
@@ -2,12 +2,40 @@ import React, { PropTypes } from 'react';
 import FontAwesome from 'react-fontawesome';
 import { Alert } from 'react-bootstrap';
 
+export const ASYNC_STATUS = Object.freeze({
```

Comment:
I prefer using the standar dot syntax for constants like enums. `immutable.js` is great for things that I need to mutate conceptually, e.g. `foo.set('bar',5)`. It does transformations like that in a really efficient manner, plus `mergeDeep`, `getIn` and other helper functions are awesome. 

Since the enumeration never needs to change, using `ASYNC_STATUS.ERROR` is cleaner and has more IDE support than `ASYNC_STATUS.get('ERROR')`